runconfig/parse.go

```diff
@@ -513,3 +505,31 @@ func ParseDevice(device string) (DeviceMapping, error) {
 	}
 	return deviceMapping, nil
 }
+
+// ParseVolumesIntoBinds parses any volumes passed by the client and moves any
+// volumes which are in Config.Volumes into HostConfig.Binds.
+func ParseVolumesIntoBinds(c *Config, hc *HostConfig) (*Config, *HostConfig, error) {
+	if c != nil && hc != nil {
+		newVolumes := c.Volumes
+		for bind := range newVolumes {
+			var (
```

Comment:
Still not entirely sure of your comment, but having looked at it more this morning, it doesn't make sense to have this in a separate function as there is only one caller, so in-lining it into the caller in DecodeContainerConfig and unifying. Will be in next push.